src/main/java/io/deepstream/Record.java

```diff
@@ -509,7 +509,9 @@ Record whenReady(RecordReadyListener recordReadyListener) {
         if( this.isReady ) {
             recordReadyListener.onRecordReady( this.name, this );
         } else {
-            this.onceRecordReadyListeners.add( recordReadyListener );
+            synchronized (this) {
```

Comment:
Without this all record ready listeners are not added properly. So when the read comes back  some are null. By adding them sync, they can all get added